### ISSUE TITLE
feat: Rubric-based error extraction via FunctionGemma

### ISSUE THREAD

**Issue by inanna-malick** [tidepool-heavy-industries/exomonad#388]
## Summary

Implement a semantic error extraction system using FunctionGemma (or Haiku for training data collection) that scans build/script output and extracts structured signals via a predefined rubric.

## Motivation

Scripts, justfiles, cabal builds, docker builds, etc. produce unstructured error output that's tedious to parse programmatically. A small model trained on this domain can extract structured signals we can act on.

### The Rubric

Predefined vocabulary of error signals:

```haskell
data ErrorFlag
  = TypeError
  | SyntaxError  
  | MissingModule Text    -- extracted module name
  | MissingDep Text       -- extracted dependency name
  | PermissionDenied
  | NetworkError
  | FileNotFound Text
  | OutOfMemory
  | Timeout
  | Suggestion Text       -- best-effort "try X" extraction
```

### The Pattern

1. **Chunk** output into ~4K token pieces (for mid-length output)
2. **Extract** flags from each chunk via `invokeLLM` effect
3. **Accumulate** flags across chunks (union)
4. **Classify** final error type from accumulated flags (pure Haskell)

```
stderr (mid-length)
    ↓ chunk
[Chunk₁, Chunk₂]
    ↓ map extractFlags (parallelizable)  
[{TypeError, MissingModule=Foo}, {Suggestion=...}]
    ↓ fold (union)
{TypeError, MissingModule=Foo, Suggestion=...}
    ↓ classify (pure)
ModuleMissing "Foo"
```

### FunctionGemma Prompt Format

```
Scan this build output for error signals.

RUBRIC (emit any that apply):
- TypeError           : type mismatch, couldn't match expected/actual
- SyntaxError         : parse error, unexpected token
- MissingModule=X     : "Could not find module X"
- MissingDep=X        : package not in cabal/package.yaml
- PermissionDenied    : EACCES, permission denied
- NetworkError        : connection refused, timeout, DNS
- FileNotFound=X      : no such file, ENOENT
- OutOfMemory         : heap exhausted, OOM
- Timeout             : exceeded time limit
- Suggestion=X        : any "try doing X" or "perhaps you meant"

Output only flags that appear. Include extracted strings where noted.

CHUNK:
[output here]
```

Output example:
```
TypeError
MissingModule=Data.Aeson
Suggestion=add aeson to build-depends
```

### Parsing

Simple line-based parsing - each line is a flag, `=` separates key from extracted string.

### Phase 1: InvokeLLM Effect
- [ ] Add `InvokeLLM` effect to `dsl/core/src/ExoMonad/Effects/`
- [ ] Implement Haiku interpreter in `effects/llm-interpreter/`
- [ ] Wire to control-server handler stack

### Phase 2: Error Extractor
- [ ] Define `ErrorFlag` type and rubric prompt
- [ ] Implement `extractFlags :: Text -> Eff es [ErrorFlag]`
- [ ] Implement chunking for mid-length output
- [ ] Implement `classifyFromFlags :: [ErrorFlag] -> BuildError`

### Phase 3: Integration
- [ ] Hook into build failure paths in control-server
- [ ] Log extractions for training data collection
- [ ] Surface classification in hook responses / MCP tool results

## Training Loop

When classification is wrong and human corrects:
1. Save chunk + extracted flags + human correction
2. Use as training data for FunctionGemma fine-tuning
3. Start with Haiku, graduate to FunctionGemma once we have data

## Inspired By

"Morphic Intelligence: Neuro-Symbolic Recursion Schemes Using FunctionGemma" - LLM as semantic ALU for local pattern matching within deterministic control flow.

## Labels

enhancement, llm, control-server
